core/qubesutils.py

```diff
@@ -469,7 +469,7 @@ def block_detach_all(vm):
 ####### USB devices ######
 
 usb_ver_re = re.compile(r"^(1|2)$")
-usb_device_re = re.compile(r"^[0-9]+-[0-9]+(_[0-9]+)?$")
+usb_device_re = re.compile(r"^[0-9]+-[0-9]+([_0-9]+)?$")
```

Comment:
`_` was intentionally outside of [], to not allow it as the last character. The proper fix is to change last `?` to `*`.